chain-abci/src/app/mod.rs

```diff
@@ -68,7 +69,9 @@ impl<T: EnclaveProxy> abci::Application for ChainNodeApp<T> {
         let mut resp = ResponseInfo::new();
         resp.app_version = chain_core::APP_VERSION;
         resp.version = get_version();
-        if let Some(app_state) = &self.last_state {
+        if let Some(raw) = chain_storage::get_last_app_state(&kv_store!(self)) {
```

Comment:
```suggestion
        if let Some(raw) = chain_storage::get_last_app_state(&self.storage) {
```
`kv_store!` is the storage combined with in-memory buffer.